src/java/TransactionReceivedListener.java

```diff
@@ -6,34 +6,44 @@
 /**
  * Created by snakecharmer1024 on 7/19/16.
  */
+
+/** TransactionReceivedListener inherits from PreMessageReceived
+ *    class from bitcoinj libraries.
+*/  
+
 public class TransactionReceivedListener implements PreMessageReceivedEventListener {
 
-    public Sha256Hash targetTxHash;
+    public Sha256Hash targetTxHash; //single class property
 
 
     public TransactionReceivedListener(Transaction tx) {
-        this(tx.getHash());
-    }
+        this(tx.getHash()); //on transaction received, 
+    } // calls itself with a different data type(hash of the transaction originally passed in)?
+
+
+    // why seperate the logic? will the second listener instantiate on listening too?
 
     public TransactionReceivedListener(Sha256Hash hash) {
         this.targetTxHash = hash;
     }
 
+
+
     public Message onPreMessageReceived(Peer peer, Message message)
     {
-        if (message instanceof InventoryMessage)
+        if (message instanceof InventoryMessage) // InventoryMessage class indentifies if message is of relevant transaction/block data
         {
-            InventoryMessage invMessage = (InventoryMessage) message;
-            for (InventoryItem item : invMessage.getItems())
+            InventoryMessage invMessage = (InventoryMessage) message; //set invMessage to message as a class instance of Inventory Message?
+            for (InventoryItem item : invMessage.getItems()) // for loop of each item in invMessage.
             {
-                if (item.hash == this.targetTxHash)
+                if (item.hash == this.targetTxHash) // does this item have the same hash as the transaction we are listening for?
                 {
-                    PermutationManager.getPermutationManager().addPeerToTransaction(peer, item.hash);
-                    break;
+                    PermutationManager.getPermutationManager().addPeerToTransaction(peer, item.hash); //then add the peer and tx to the transactionPermutationHashMap
+                    break; 
                 }
             }
         }
-        return message;
+        return message; // otherwise return the message, is this needed?
```

Comment:
Not sure, but all EventListeners (?) return a message, this gives us an opportunity to modify it before it is processed by downstream listeners. We don't actually modify it, just log it.